### ISSUE TITLE
Error during eslint --init: "EACCES: permission denied, scandir […]"

### ISSUE THREAD

**Issue by quinncomendant** [eslint/eslint#6498]
**What version of ESLint are you using?**
v2.11.1

**What parser (default, Babel-ESLint, etc.) are you using?**
I don't know.

**Please show your full configuration:**
I haven't created one.

**What did you do? Please include the actual source code causing the issue.**
I'm just getting started using `eslint` on the command line. 
1. I installed it with `npm -g install eslint`
2. then ran `eslint --init`
3. I filled in the options like this:
   
   ? How would you like to configure ESLint? Inspect your JavaScript file(s)
   ? Which file(s), path(s), or glob(s) should be examined? /Users/q/src/fwr/cp/assets/js/orders.js 
   ? What format do you want your config file to be in? YAML
   ? Are you using ECMAScript 6 features? No
   ? Where will your code run? Browser
   ? Do you use CommonJS? No
   ? Do you use JSX? No

**What did you expect to happen?**
I expected it to examine the file at `/Users/q/src/fwr/cp/assets/js/orders.js` and generate a config file.

**What actually happened? Please include the actual, raw output from ESLint.**
It appears to try to access a file under `/Library/Application Support/Apple/ParentalControls/Users`  (which exists, but is owned by `root` and has `drwx------` perms). ESLint dies with this error:

```
Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s 

EACCES: permission denied, scandir '/Library/Application Support/Apple/ParentalControls/Users'
Error: EACCES: permission denied, scandir '/Library/Application Support/Apple/ParentalControls/Users'
    at Error (native)
    at Object.fs.readdirSync (fs.js:945:18)
    at GlobSync._readdir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:275:41)
    at GlobSync._readdirInGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:254:20)
    at GlobSync._readdir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:263:17)
    at GlobSync._processReaddir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:135:22)
    at GlobSync._process (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:130:10)
    at GlobSync._processGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:367:10)
    at GlobSync._process (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:128:10)
    at GlobSync._processGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:370:10)
```

**Comment by quinncomendant** [comment]
FYI, using the `Answer questions about your style` option when answering `How would you like to configure ESLint?` does work. I've been able to generate a `.eslintrc.yml` file that way.

**Comment by nzakas** [comment]
ESLint doesn't have access to that directory. You either need to give access to the user running ESLint, or run ESLint as root.

**Comment by quinncomendant** [comment]
Obviously, ESLint doesn't have permission to access `/Library/Application Support/Apple/ParentalControls/Users`. The question is: why was it trying to access that directory, when the path I gave it to analyze was `/Users/q/src/fwr/cp/assets/js/orders.js` (which it does have permission to access)?

In other words, where did ESLint even get the idea to access `/Library/Application Support/Apple/ParentalControls/Users`? Never did I enter that path into ESLint, or anywhere else in my environment (`env | grep ParentalControls` shows nothing). I didn't even know that path existed. My `cwd` was `/Users/q/src/fwr`. 

Perhaps a screenshot will explain more clearly how I executed ESLint:
<img width="824" alt="screen shot 2016-06-22 at 19 12 08" src="https://cloud.githubusercontent.com/assets/6884276/16288031/6268a486-38ad-11e6-8905-5f4c2ebc000b.png">

Weird, isn't it?

**Comment by nzakas** [comment]
I have no idea. Do you have any symlinks to that directory? You can try running with `--debug` and see if there's any insights there. 

@ianvs any thoughts?

**Comment by IanVS** [comment]
I have a theory, but running with `--debug` would help a lot.

My theory is this.  `eslint --init` assumes that it's being run from the root of your project (otherwise how would it know where to put the generated config file?).  My guess is that because you provided an absolute path, for some reason it's trying to traverse all the directories in your root, and that one was the first that it didn't have read permissions to.  I'll try to figure out if and why it's doing that (which is where debug info would come in handy). 

That's my theory, but I can't confirm it because so far I'm not able to reproduce the failure.  Any other relevant info you can give about your directory structure would also be helpful.

**Comment by quinncomendant** [comment]
> My guess is that because you provided an absolute path, for some reason it's trying to traverse all the directories in your root, and that one was the first that it didn't have read permissions to.

That is a darn fine theory. Here's a PoC that shows that `/Library/Application Support/Apple/ParentalControls/Users` is the first non-readable file in the directory tree that isn't under a *dot-file* or */dev*:

    [q@x/0 ~] find /Library -name somedeepfile -ls 2>&1
    find: /.DocumentRevisions-V100: Permission denied
    find: /.fseventsd: Permission denied
    find: /.Spotlight-V100: Permission denied
    find: /.Trashes: Permission denied
    find: /dev/fd/3: Not a directory
    find: /dev/fd/4: Not a directory
    find: /Library/Application Support/Apple/ParentalControls/Users: Permission denied
    […]

(Maybe we can assume that whatever is doing the transversing will skip dot-files and things under /dev.)

Anyways, I can't reproduce the problem now. It is successfully scanning the file I give it and generating a `.eslintrc.yml` I wonder if this only occurred on a first-run because the transversing function scanned the file system and cached the results, so it isn't scanning again.

**Comment by IanVS** [comment]
@quinncomendant has this problem ever cropped up again for you?  

If not, I think maybe we can close the issue and do some more investigating if we see it pop up again.

**Comment by vitorbal** [comment]
friendly ping @quinncomendant any news on this?

**Comment by quinncomendant** [comment]
Yes, it's still a problem. I've just updated all the things (`brew update && brew upgrade && npm update -g`) and run eslint (v3.3.1) from a virgin directory (project never used eslint before). Here's output with `--debug`:

	[q@x/0 backfeed] eslint --init --debug
	? How would you like to configure ESLint? Inspect your JavaScript file(s)
	? Which file(s), path(s), or glob(s) should be examined? /Users/q/src/backfeed/global.js 
	? What format do you want your config file to be in? JavaScript
	? Are you using ECMAScript 6 features? No
	? Where will your code run? Browser
	? Do you use CommonJS? No
	? Do you use JSX? No
	Determining Config: 0% [------------------------------] 0.0s elapsed, eta 0.0s   eslint:source-code-util constructed options: +0ms { configFile: null,
	  baseConfig: { rules: {}, env: { browser: true } },
	  rulePaths: [],
	  useEslintrc: false,
	  envs: [],
	  globals: [],
	  rules: {},
	  extensions: [ '.js' ],
	  ignore: true,
	  ignorePath: null,
	  parser: '',
	  cache: false,
	  cacheLocation: '',
	  cacheFile: '.eslintcache',
	  fix: false,
	  allowInlineConfig: true,
	  cwd: '/Users/q/src/backfeed' }
	  eslint:ignored-paths Looking for ignore file in /Users/q/src/backfeed +33ms
	  eslint:ignored-paths Could not find ignore file in cwd +1ms
	  eslint:glob-util Creating list of files to process. +1ms


	EACCES: permission denied, scandir '/Library/Application Support/Apple/ParentalControls/Users'
	Error: EACCES: permission denied, scandir '/Library/Application Support/Apple/ParentalControls/Users'
	    at Error (native)
	    at Object.fs.readdirSync (fs.js:951:18)
	    at GlobSync._readdir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:276:41)
	    at GlobSync._readdirInGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:255:20)
	    at GlobSync._readdir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:264:17)
	    at GlobSync._processReaddir (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:136:22)
	    at GlobSync._process (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:131:10)
	    at GlobSync._processGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:368:10)
	    at GlobSync._process (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:129:10)
	    at GlobSync._processGlobStar (/usr/local/lib/node_modules/eslint/node_modules/glob/sync.js:371:10)

I noticed it pauses for a long time (27 sec) after the `Creating list of files to process` step. This is probably because it's searching my entire filesystem, rather than the very small number of children below the current working directory. Fixing this "permissions" problem will also fix this delay and speed up the init process.

**Comment by kaicataldo** [comment]
I'm closing this issue because it looks like consensus couldn't be reached. While we wish we could accommodate all requests, we have limited resources and do need to prioritize. We've found that issues failing to reach consensus after 21 days tend to never reach consensus, and as a team have decided to close such issues. This doesn't mean the idea isn't interesting or valuable, only that it's not something the team can commit to at the moment.
